metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -270,6 +272,39 @@ object MetalsEnrichments
       FileIO.slurp(path, StandardCharsets.UTF_8)
     }
 
+    def isValidClasspathEntry(): Boolean =
+      path.isDirectory || hasZipMagicNumbers()
+
+    /**
+     * Returns true if this file can be read as a ZIP/Jar file.
+     *
+     * It's not sufficient to check the filename extension because `*.jar`
+     * files can contain non-ZIP contents causing `java.util.zip.ZipError`
+     * which is difficult to recover from.
+     */
+    def hasZipMagicNumbers(): Boolean = {
```

Comment:
```scala
private val zipMagicNumbers = Set(0x504B0304, 0x504B0506, 0x504B0708)
def hasZipMagicNumbers(): Boolean = {
  val in = new DataInputStream(Files.newInputStream(path))
  try {
    val number = in.readInt()
    zipMagicNumbers.contains(number)
  } catch {
    case NonFatal(e) => false
  } finally {
    in.close()
  }
}
```

Reply:
This method would crash if `path` doesn't exist
```
@ java.nio.file.Files.newInputStream(java.nio.file.Paths.get("doesnotexist"))
java.nio.file.NoSuchFileException: doesnotexist
```
I tried to avoid the double try/catch but wasn't able to make it look clean. In the end, this method is self-contained enough I don't think minor improvements in readability will make a big difference in maintainability.